extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -276,22 +276,45 @@ export class InstrumentationService {
   }
 
   /**
+   * @param {!Element} el
+   * @private
+   * @return {Object}
+   */
+  extractAnalyticsVariables_(el) {
+    const dataSet = el.dataset;
+    const vars = 'vars';
+    const analyticsVariables = {};
+    for (const dataKey in dataSet) {
+      if (dataKey.indexOf(vars) === 0) {
+        const variableName = dataKey.toLowerCase().replace(vars,'');
+        analyticsVariables[variableName] = dataSet[dataKey];
+      }
+    }
+    return analyticsVariables;
+  }
+
+  /**
    * @param {!Function} listener
    * @param {string} selector
    * @private
    */
   createSelectiveListener_(listener, selector) {
     return e => {
+      let el = e.target;
+      let dataSet = null;
       // First do the cheap lookups.
-      if (selector === '*' || this.matchesSelector_(e.target, selector)) {
-        listener(new AnalyticsEvent(AnalyticsEventType.CLICK));
+      if (selector === '*' || this.matchesSelector_(el, selector)) {
+        // See https://github.com/ampproject/amphtml/issues/1298
```

Comment:
you don't need this comment. We can find the details using blame history of this line.